core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -406,9 +417,16 @@ class RemoteIndexCacheTest {
   @Test
   def testReloadCacheAfterClose(): Unit = {
     // close existing cache created in test setup before creating a new one
+    val cacheForEstimate = new RemoteIndexCache(2L, rsm, tpDir.toString)
+    val tpIdForEstimate = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
+    val metadataListForEstimate = generateRemoteLogSegmentMetadata(size = 1, tpIdForEstimate)
+    val entryForEstimate = cacheForEstimate.getIndexEntry(metadataListForEstimate.head)
+    val estimateEntryBytesSize = RemoteIndexCache.estimatedEntrySize(entryForEstimate)
+    Utils.closeQuietly(cacheForEstimate, "RemoteIndexCache created for estimating entry size")
```

Comment:
do we need to do all this? Instead can't we initialize with a big enough size that works for this test case?

Reply:
Because this test will test the size reserved for only two elements, that is, put three in, it will be verified that since there are only two in the eviction, the cache here cannot be set large enough. @divijvaidya